File: apps/go/application/fizz_buzz_list_command.go
package application

import (
	"github.com/k2works/getting-started-tdd/apps/go/domain/model"
	type_ "github.com/k2works/getting-started-tdd/apps/go/domain/type_"
)

// FizzBuzzListCommand は FizzBuzzList を生成するコマンドです。
type FizzBuzzListCommand struct {
	count        int
	fizzBuzzType type_.FizzBuzzType
}

// NewFizzBuzzListCommand は FizzBuzzListCommand を生成します。
func NewFizzBuzzListCommand(fizzBuzzType type_.FizzBuzzType, count int) *FizzBuzzListCommand {
	return &FizzBuzzListCommand{count: count, fizzBuzzType: fizzBuzzType}
}

// Execute はリスト生成コマンドを実行し FizzBuzzList を返します。
func (c *FizzBuzzListCommand) Execute() interface{} {
	values := make([]model.FizzBuzzValue, c.count)
	for i := 0; i < c.count; i++ {
		values[i] = c.fizzBuzzType.Generate(i + 1)
	}
	return model.NewFizzBuzzList(values)
}

File: apps/rust/src/application/fizz_buzz_list_command.rs
use crate::domain::model::FizzBuzzList;
use crate::domain::types::FizzBuzzType;
use crate::fizz_buzz::create;

use super::FizzBuzzCommand;

pub struct FizzBuzzListCommand {
    fizz_buzz_type: Box<dyn FizzBuzzType>,
}

impl FizzBuzzListCommand {
    pub fn new(type_number: i32) -> Result<Self, String> {
        let fizz_buzz_type = create(type_number)?;
        Ok(Self { fizz_buzz_type })
    }
}

impl FizzBuzzCommand for FizzBuzzListCommand {
    fn execute(&self) -> String {
        let list = FizzBuzzList::new(self.fizz_buzz_type.as_ref());
        list.to_string()
    }
}
